### ISSUE TITLE
The program crashes after I press "Win + D"

### ISSUE THREAD

**Issue by Jimmie00x0000** [nmwsharp/polyscope#36]
System: Win 10
polyscope version: latest

I often use "Win + D" to show the desktop, but the polyscope program crashes after I press this shortcut. Is this a bug?
I use the following code to show the GUI:
```c++
polyscope::init();
polyscope::show();
```

**Comment by nmwsharp** [comment]
I haven't seen this before, but it sounds like it might be a GLFW-related bug.

Unfortunately I'm away from my Windows machine for the next few weeks, so it might be a while until I can debug this.

If you don't want to wait and are willing to do some tests, we might be able to figure it out from here. Could you run in Debug mode / with a debugger to see where the crash happens?

**Comment by nmwsharp** [comment]
FYI, was able to reproduce this on a coworker's Windows machine today. 

There is indeed a crash on Windows if you minimize the Polyscope window, because it tries to render when there is no valid context.

The crash occurs around here: https://github.com/nmwsharp/polyscope/blob/master/src/gl/gl_utils.cpp#L232

I don't know a fix yet; need to figure why Windows is different from other platforms in this behavior. Probably some missing GLFW call.

**Comment by Jimmie00x0000** [comment]
> FYI, was able to reproduce this on a coworker's Windows machine today.
> 
> There is indeed a crash on Windows if you minimize the Polyscope window, because it tries to render when there is no valid context.
> 
> The crash occurs around here: https://github.com/nmwsharp/polyscope/blob/master/src/gl/gl_utils.cpp#L232
> 
> I don't know a fix yet; need to figure why Windows is different from other platforms in this behavior. Probably some missing GLFW call.

I have a simple fix. If framebuffer status is not okay, just early return from renderScene(). It seems to work fine:
```c++
// in gl_utils.cpp 
// change the return type to bool
bool GLFramebuffer::bindForRendering() {
  glBindFramebuffer(GL_FRAMEBUFFER, handle);

  // Check if the frame buffer is okay
  if (glCheckFramebufferStatus(GL_FRAMEBUFFER) != GL_FRAMEBUFFER_COMPLETE) {
    // log something ...
    return false;
    // throw std::runtime_error("OpenGL error occurred: framebuffer not complete!");
  }

  // .....
  // ....
  // ...
  return true;
}

// in polyscope.cpp
void renderScene() {

  // Activate the texture that we draw to
  sceneFramebuffer->resizeBuffers(view::bufferWidth, view::bufferHeight);
  sceneFramebuffer->setViewport(0, 0, view::bufferWidth, view::bufferHeight);
  if (!sceneFramebuffer->bindForRendering()) return;
  sceneFramebuffer->clearColor = {view::bgColor[0], view::bgColor[1], view::bgColor[2]};
  sceneFramebuffer->clearAlpha = 0.0;
  sceneFramebuffer->clear();

  // If a view has never been set, this will set it to the home view
  view::ensureViewValid();

  drawStructures();

  // Draw the ground plane
  gl::drawGroundPlane();
}
```

**Comment by nmwsharp** [comment]
Whoops, this had slipped my mind. Should be fixed in 891ce86c4842c364c0821a7ffa4f0bf8f4875bae. Thank you for reporting and following up!
